### PR TITLE
Various cleaning

### PR DESCRIPTION
**Pull request by presidentJFK**
Here are some changes I have made to my local after_do while I have been using and reading through the code, that might help the next developer.

A summary of my changes is as follows:

* refactored so the _after_do_aliased_name is only called once
* I have changed some wording in comments that confused me when I first read through
* added some extra comments around some of the examples
* removed a couple extra spaces I saw

### CODE REVIEW

**Comment by stupied4ever** [comment]
hi @presidentJFK, thx a lot for your effort, it looks nice, except one little thing. 
You changed some single quotes by double quotes, and thats out of the ruby style guide.

if you fix that, I will be glad to merge.

**Comment by presidentJFK** [comment]
@stupied4ever thanks for looking and I updated that commit to only use double quotes for the interpolation.

File: lib/after_do.rb [hunk 1]
```diff
@@ -60,17 +61,19 @@ def _after_do_methods_hash
...
+    unless _after_do_already_redefined?(method, alias_name)
+      _after_do_redefine_method(method, alias_name)
+    end
     @_after_do_callbacks[type][method] << block
   end
 
-  def _after_do_already_redefined?(method)
-    private_instance_methods(false).include? _after_do_aliased_name(method)
+  def _after_do_already_redefined?(method, alias_name)
+    private_instance_methods(false).include? alias_name
```
Comment (PragTob) [inline lib/after_do.rb]: the `method` argument is unused now, will fix that up

File: samples/getting_a_hold.rb [hunk 2]
```diff
@@ -17,10 +17,24 @@ def value
 Example.extend AfterDo
 
 Example.after :zero do puts 'Hello!' end
+
+# If the callback method takes no arguments
+# then the first argument passed to the block
+# with will the instance of the class you are in.
```
Comment (PragTob) [inline samples/getting_a_hold.rb]: I guess it is `will be the instance` ?

File: samples/getting_a_hold.rb [hunk 3]
```diff
@@ -17,10 +17,24 @@ def value
...
+
+# Callback methods can take arguments.
 Example.after :two do |first, second| puts first + ' ' + second end
+
+# If the callback takes arguments, the last argument, after the required
+# arguments will be the instance of the class.
 Example.after :two do |a, b, obj| puts a + ' ' + b + ' ' + obj.value end
-Example.after :two do |*, obj| puts 'just ' +  obj.value end
+
+# You can also you use the * to suck up all the required arguments
```
Comment (PragTob) [inline samples/getting_a_hold.rb]: double you :)

**Comment by PragTob** [comment]
Thanks for the improvements :+1: 

I'll patch a couple of things up and then push it out. Sorry for the build failure, rubinius seems to be broken somehow :)

@stupied4ever thanks for the reviewing help, I don't believe you got merge rights though :) Also on strings, my stance atm is that it doesn't matter as there are valid points for both the`'` and the `"` style :)

File: samples/getting_a_hold.rb [hunk 4]
```diff
@@ -30,4 +44,5 @@ def value
 # some value
 # one two
 # one two some value
-# just some value
\ No newline at end of file
+# args passed to callback: one, two
```
Comment (PragTob) [inline samples/getting_a_hold.rb]: oops, good catch!

**Comment by PragTob** [comment]
Thanks for sharing your improvements @presidentJFK ! :+1:

**Comment by stupied4ever** [comment]
@PragTob my bad, I just got confused with my other project (https://github.com/stupied4ever/after_do-logging), as I did not commit there for a while, I just confused the name.

About single or double quotes, the problem is that the other project follow ruby stile guide most as possible. And As I confused both projects, I was thinking as it was on that project. (because of that I sad I would merge).

Sorry by the mess.

**Comment by PragTob** [comment]
@stupied4ever Yeah I figured, also remembered that I wanted to look at that project (and the loader) some time and then forgot, took a look, looked good :) Also noticed that you use after_do in another way that I usually do - which means that it's great that it allows it :+1: Also means, I probably won't implement an `Aspect` module/class or something as it seems to be specific to the way I use after_do :)

**Pull request merged**